js/adapt-stateful-session.js

```diff
@@ -61,29 +61,41 @@ define([
       var sessionPairs = Adapt.offlineStorage.get();
       var hasNoPairs = _.keys(sessionPairs).length === 0;
 
+      var courseState;
       var doSynchronousPart = function() {
-        if (sessionPairs.questions && this._shouldStoreResponses) questions.deserialize(sessionPairs.questions);
-        if (sessionPairs._isCourseComplete) Adapt.course.set('_isComplete', sessionPairs._isCourseComplete);
-        if (sessionPairs._isAssessmentPassed) Adapt.course.set('_isAssessmentPassed', sessionPairs._isAssessmentPassed);
+        if (sessionPairs.q && this._shouldStoreResponses) questions.deserialize(sessionPairs.q);
+        if (courseState) Adapt.course.set('_isComplete', courseState[0]);
+        if (courseState) Adapt.course.set('_isAssessmentPassed', courseState[1]);
         callback();
       }.bind(this);
 
       if (hasNoPairs) return callback();
 
+      if (sessionPairs.c) {
+        courseState = SCORMSuspendData.deserialize(sessionPairs.c);
+      }
+
       // Asynchronously restore block completion data because this has been known to be a choke-point resulting in IE8 freezes
-      if (sessionPairs.completion) {
-        serializer.deserialize(sessionPairs.completion, doSynchronousPart);
+      if (courseState) {
+        serializer.deserialize(courseState.slice(2).map(Number).map(String).join(''), doSynchronousPart);
```

Comment:
> this has been known to be a choke-point resulting in IE8 freezes

Is there a reason we're still doing this even though we stopped supporting IE8 some time ago? If it's just because there are performance concerns with IE11 still, let's update the comment to make that clear.

Reply:
i didn't want to change it because it works and ie11 testing is a bit too big for this pr, will add a comment as such